### PR TITLE
feat: Add etcd ca generation to userdata.Generate

### PR DESCRIPTION
**Pull request by bradbeam**
Allow us to generate/manage tls assets external to kubeadm.

Signed-off-by: Brad Beam <brad.beam@talos-systems.com>

### CODE REVIEW

File: pkg/userdata/generate/generate.go [hunk 1]
```diff
@@ -282,6 +284,22 @@ func NewInput(clustername string, masterIPs []string, kubernetesVersion string)
 	if err != nil {
 		return nil, err
 	}
+	caPemBlock, _ := pem.Decode(osCert.CrtPEM)
```
Comment (bradbeam) [inline pkg/userdata/generate/generate.go]: Most of the red here is just grouping things up a little better

File: pkg/userdata/generate/controlplane.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ security:
     ca:
       crt: "{{ .Certs.K8sCert }}"
       key: "{{ .Certs.K8sKey }}"
+    etcd:
```
Comment (Ulexus) [inline pkg/userdata/generate/controlplane.go]: I know this is all kind of temporary, but etcd should not be under kubernetes.  Both topologically and from a PKI standpoint, it should be a separate top-level category.

Further, etcd has two PKI groups:  peering and client-server, and even if we use the same for both right now, the userdata should reflect these distinctions (even if it is just a subcategorization).

So, for instance, I would use security.etcd.peer.ca and security.etcd.service.ca and default the service PKI to peer, if it is not defined.

File: pkg/userdata/generate/generate.go [hunk 3]
```diff
@@ -313,34 +331,31 @@ func NewInput(clustername string, masterIPs []string, kubernetesVersion string)
...
 	if err != nil {
 		return nil, err
 	}
 
 	certs := &Certs{
 		AdminCert: base64.StdEncoding.EncodeToString(adminCrt.X509CertificatePEM),
 		AdminKey:  base64.StdEncoding.EncodeToString(adminKey.KeyPEM),
-		OsCert:    base64.StdEncoding.EncodeToString(osCert.CrtPEM),
-		OsKey:     base64.StdEncoding.EncodeToString(osCert.KeyPEM),
+		EtcdCert:  base64.StdEncoding.EncodeToString(etcdCert.CrtPEM),
```
Comment (Ulexus) [inline pkg/userdata/generate/generate.go]: Same as above:  EtcdPeerCert, EtcdServiceCert

File: pkg/userdata/v1alpha1/cluster_config.go [hunk 4]
```diff
@@ -55,6 +55,8 @@ type SchedulerConfig struct {
 // EtcdConfig represents etcd config vals
 type EtcdConfig struct {
 	Image string `yaml:"image,omitempty"`
+	Crt   string `yaml:"crt"`
```
Comment (Ulexus) [inline pkg/userdata/v1alpha1/cluster_config.go]: It is unclear that this is the CA cert, not the peer, service, or client certs (assuming it _is_ the CA cert, of course).

File: pkg/userdata/generate/controlplane.go (hunk 2 above)
Comment (andrewrynhard) [reply pkg/userdata/generate/controlplane.go]: +1

Comment (bradbeam) [reply pkg/userdata/generate/controlplane.go]: think this is addressed in the latest commit. 
```
security.etcd.ca.key // security.etcd.ca.crt
```

File: pkg/userdata/generate/generate.go (hunk 3 above)
Comment (bradbeam) [reply pkg/userdata/generate/generate.go]: the above is just internal, I'm not sure it matters much ( ex, we don't really distinguish between the OS level certs ( osd )).

**Pull request merged**